### PR TITLE
Implemented Redshift dialect

### PR DESCRIPTION
**Pull request by abourget**
plus some --help improvements..

Each commit can be cherry-picked independently if only the redshift dialect is wanted for example.

### CODE REVIEW

File: cmd/goose/cmd_dbversion.go [hunk 1]
```diff
@@ -9,9 +9,10 @@ import (
 
 var dbVersionCmd = &Command{
 	Name:    "dbversion",
-	Usage:   "",
+	Usage:   "dbversion",
 	Summary: "Print the current version of the database",
-	Help:    `dbversion extended help here...`,
+	Help:    `dbversion introspects the database and prints out the last migration
```
Comment (zabawaba99) [inline cmd/goose/cmd_dbversion.go]: Perhaps something like `dbVersion prints out the last migration applied` might be a little more clear.

**Comment by phemmer** [comment]
Thanks for the PR. I tested it and everything appears to be fully functional.

I'd be good with merging this if you can remove all the changes in the `cmd` directory (the help stuff). Basically just drop commit f5556b1. These changes have issues, but are also out of scope for this PR. If you still want them merged in, you're welcome to open another PR specifically for them and we can discuss them there.

**Comment by abourget** [comment]
Reworked the PR, this implements the Redshift dialect only.. with an additional recent bugfix.

**Comment by phemmer** [comment]
LGTM

**Pull request merged**